javalib/src/main/scala/java/io/CharArrayReader.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.io
+
+class CharArrayReader(protected var buf: Array[Char], offset: Int, length: Int) extends Reader {
+  if (offset < 0 || offset > buf.length || length < 0 || offset + length < 0) throw new IllegalArgumentException
+
+  protected var pos: Int = offset
+  protected var markedPos: Int = offset
+  protected var count: Int = {
+    val bufferLength = buf.length
+
+    if (offset + length < bufferLength) length
```

Comment:
This line is suspicious. Shouldn't all of this just be:

```scala
count: Int = Math.min(buf.length, offset + length)
```

If yes, this points to a weakness in the tests.